### PR TITLE
feat: added usecase generator

### PR DESCRIPTION
**Pull request by fkrjakDatasentics**
Use case generator for table added 

Example notebook https://adb-8406481409439422.2.azuredatabricks.net/?o=8406481409439422#notebook/1872984241537364/command/3964698646820817

### CODE REVIEW

File: src/odap/use_case/generate.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from pyspark.sql import SparkSession
+from pyspark.sql.utils import AnalysisException
+from odap.common.config import get_config_namespace, ConfigNamespace
+from odap.use_case.schemas import get_use_case_schema
+from odap.use_case.usecases import generate_usecases
+
+
+def generate_table():
+    spark = SparkSession.getActiveSession()
+    path = get_config_namespace(ConfigNamespace.USECASE_FACTORY)["path"]
```
Comment (ds-lukaslangr) [inline src/odap/use_case/generate.py:10]: shouldn't this be called `table_name` instead?

File: src/odap/use_case/generate.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    data = spark.createDataFrame(data=generate_usecases(), schema=get_use_case_schema())
+
+    try:
```
Comment (ds-lukaslangr) [inline src/odap/use_case/generate.py:13]: you probably shouldn't use try except for control flow
this whole thing could be handled using [delta merge](https://docs.delta.io/latest/delta-update.html#merge-examples) where `name` is the primary key of the use case table

or why not just overwrite the table everytime?

File: src/odap/use_case/notebooks.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    content = output["content"]
+    return b64decode(content).decode("utf-8").split("COMMAND ----------")[1]
+
+
+def get_segment_attributes(segment: str) -> List[str]:
+    segments = re.match(
+        "SELECT (.*)FROM",
+        get_segment_data(segment).replace("\n", ""),
+    )
+    if segments:
```
Comment (ds-lukaslangr) [inline src/odap/use_case/notebooks.py:22]: switch this around so it's `if not segments`

**Review by ds-lukaslangr — CHANGES_REQUESTED** [review]

File: src/odap/use_case/generate.py (hunk 1 above)
Comment (fkrjakDatasentics) [reply src/odap/use_case/generate.py:10]: Sounds better than my name, changing it

File: src/odap/use_case/generate.py (hunk 2 above)
Comment (fkrjakDatasentics) [reply src/odap/use_case/generate.py:13]: True

**Review by ds-lukaslangr — APPROVED** [review]

**Pull request closed without merging**